hcl/hclsyntax/token.go

```diff
@@ -223,16 +228,30 @@ func checkInvalidTokens(tokens Tokens) hcl.Diagnostics {
 		case TokenBacktick:
 			// Only report for alternating (even) backticks, so we won't report both start and ends of the same
 			// backtick-quoted string.
-			if toldExponent < 4 && (toldExponent%2) == 0 {
+			if (toldBacktick % 2) == 0 {
 				diags = append(diags, &hcl.Diagnostic{
 					Severity: hcl.DiagError,
 					Summary:  "Invalid character",
 					Detail:   "The \"`\" character is not valid. To create a multi-line string, use the \"heredoc\" syntax, like \"<<EOT\".",
 					Subject:  &tok.Range,
 				})
-
+			}
+			if toldBacktick <= 2 {
 				toldBacktick++
 			}
+		case TokenApostrophe:
+			if (toldApostrophe % 2) == 0 {
+				newDiag := &hcl.Diagnostic{
+					Severity: hcl.DiagError,
+					Summary:  "Invalid character",
+					Detail:   "The ' character is not valid. Use double quotes (\") to enclose strings.",
```

Comment:
I worry that the literal `'` here may be hard to read or confusing... maybe instead we could say:

```
Single quotes are not valid. Use double quotes (") to enclose strings.
```

The rendered error message should highlight the single quote character in case the reader isn't sure what a single quote is, but I expect most readers will be familiar with this terminology from other programming languages.

(I didn't do that for the backtick because I was worried that the terminology "backtick" has Unix heritage and so may be less familiar to users coming from other platforms, but "single quote" feels more accessible to me, though of course that's subjective.)